pkg/image/builder.go

```diff
@@ -527,6 +527,13 @@ func IsPrivateImage(image string) (bool, error) {
 
 		sysCtx := types.SystemContext{DockerDisableV1Ping: true}
 
+		if dockerHubRegistry.Username != "" && dockerHubRegistry.Password != "" {
```

Comment:
I think you need to also check domain is docker.io.   This will get host name from parsed reference
```
		registryHost := reference.Domain(ref.DockerReference())
```